### ISSUE TITLE
Class with matching class/static methods doesn't check with `Protocol`

### ISSUE THREAD

**Issue by TTsangSC** [astral-sh/ty#3429]
### Summary

Description
----
Classes which provide methods (via static/class methods) and attributes (via `ClassVar`s) compliant to a `Protocol` interface do not type-check as instances thereof.

Example
----
(Playground: https://play.ty.dev/7a74847e-6b8e-408c-98f1-a7c1c0f6b072)

```python
from typing import Any, Protocol


class Applicable(Protocol):
    def apply(self, obj: Any) -> Any:
        ...


class Patch:
    def apply(self, obj: Any) -> None:
        ...


class PatchClass:
    @staticmethod
    def apply(obj: Any) -> None:
        ...


a1: Applicable = Patch()
a2: Applicable = PatchClass
```

Expected behavior
----
No errors.

Actual behavior
----
```python
Object of type `<class 'PatchClass'>` is not assignable to `Applicable` (invalid-assignment) [Ln 21, Col 18]
```

### Version

ty 0.0.34 (d00448eb0 2026-05-01)

**Comment by sharkdp** [comment]
Thank you for reporting this.

The full diagnostic hint that ty gives is:
```py
error[invalid-assignment]: Object of type `<class 'PatchClass'>` is not assignable to `Applicable`
  --> main.py:21:5
   |
21 | a2: Applicable = PatchClass
   |     ----------   ^^^^^^^^^^ Incompatible value of type `<class 'PatchClass'>`
   |     |
   |     Declared type
   |
info: type `<class 'PatchClass'>` is not assignable to protocol `Applicable`
info: └── protocol member `apply` is not defined on type `<class 'PatchClass'>`
```

The last line here definitely seems like a bug.

**Comment by TTsangSC** [comment]
Thanks for the update.

Apropos, I do see the full diagnostic on local, but only the terse one-liner from the Playground website. Do you happen to know if there's a toggle on web for more verbose output? Cheers.

**Comment by carljm** [comment]
I think this may be intentional behavior; it's basically a variant of #931. The diagnostic message could certainly be improved.

The puzzle is that people want class objects and module objects to satisfy protocols with methods, but people also want `type[SomeProto]` to describe a type that has the methods defined on `SomeProto`. These two are in conflict: the second assumes that methods defined on `SomeProto` must exist on the _type_ of an object matching `SomeProto`, which would mean the first should not be allowed.

One option is to do what other type checkers do, which is to be unsound and inconsistent: assume in the second case that the methods do exist on `type[SomeProto]`, but still allow objects for which that is not true to be subtypes of the protocol.

**Comment by TTsangSC** [comment]
Makes sense. Thanks for the explanation!
